patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -65,16 +65,84 @@
        }
     }
  
-@@ -553,7 +554,7 @@
+@@ -493,14 +494,11 @@
+    @Deprecated
+    public List<ItemStack> func_220076_a(BlockState p_220076_1_, LootContext.Builder p_220076_2_) {
+       ResourceLocation resourcelocation = this.func_220068_i();
+-      if (resourcelocation == LootTables.field_186419_a) {
+-         return Collections.emptyList();
+-      } else {
+-         LootContext lootcontext = p_220076_2_.func_216015_a(LootParameters.field_216287_g, p_220076_1_).func_216022_a(LootParameterSets.field_216267_h);
+-         ServerWorld serverworld = lootcontext.func_202879_g();
+-         LootTable loottable = serverworld.func_73046_m().func_200249_aQ().func_186521_a(resourcelocation);
+-         return loottable.func_216113_a(lootcontext);
+-      }
++      p_220076_2_ = p_220076_2_.func_216015_a(LootParameters.field_216287_g, p_220076_1_);
++      ServerWorld serverworld = p_220076_2_.func_216018_a();
++      LootTable loottable = serverworld.func_73046_m().func_200249_aQ().func_186521_a(resourcelocation);
++
++      return net.minecraftforge.event.ForgeEventFactory.fireBlockGenerateLootEvent(loottable, p_220076_2_);
+    }
+ 
+    public static List<ItemStack> func_220070_a(BlockState p_220070_0_, ServerWorld p_220070_1_, BlockPos p_220070_2_, @Nullable TileEntity p_220070_3_) {
+@@ -514,15 +512,18 @@
+    }
+ 
+    public static void func_220078_b(BlockState p_220078_0_, LootContext.Builder p_220078_1_) {
++      startDropCapture();
+       ServerWorld serverworld = p_220078_1_.func_216018_a();
+       BlockPos blockpos = p_220078_1_.func_216024_a(LootParameters.field_216286_f);
+       p_220078_0_.func_215693_a(p_220078_1_).forEach((p_220063_2_) -> {
+          func_180635_a(serverworld, blockpos, p_220063_2_);
+       });
+       p_220078_0_.func_215706_a(serverworld, blockpos, ItemStack.field_190927_a);
++      finishDropCapture(serverworld, blockpos, p_220078_0_, p_220078_1_.func_216022_a(LootParameterSets.field_216267_h));
+    }
+ 
+    public static void func_220075_c(BlockState p_220075_0_, World p_220075_1_, BlockPos p_220075_2_) {
++      startDropCapture();
+       if (p_220075_1_ instanceof ServerWorld) {
+          func_220070_a(p_220075_0_, (ServerWorld)p_220075_1_, p_220075_2_, (TileEntity)null).forEach((p_220079_2_) -> {
+             func_180635_a(p_220075_1_, p_220075_2_, p_220079_2_);
+@@ -530,9 +531,11 @@
+       }
+ 
+       p_220075_0_.func_215706_a(p_220075_1_, p_220075_2_, ItemStack.field_190927_a);
++      finishDropCapture(p_220075_1_, p_220075_2_, p_220075_0_, null);
```

Comment:
Regarding all of these patches, do we want to care about implementations of spawnAdditionalDrops? 
With out changes elseware, the only implementation of this that does anything is SilverfishBlock to spawn the silverfish.
Not droping any additional items. It would be a slightly more complex patch wise, but we could bypass the threadlocal drop captures by just using the return value of getDrops directly.

Reply:
Yes and no, even if vanilla doesn't do much with the spawnAdditionalDrops method that doesn't mean modders won't use it.

So if we ignored spawnAdditionalDrops altogether then there would be issues with mods spawning their drops from that method but no event being fired to actually change what those drops are.

If you'd prefer to make drop changes happen at the block state level (which seems to be where things get determined before dropping from what I've seen) then yes that could be done. I'm just concerned about edge cases and having people complain about drop events not getting fired for their blocks.

Reply:
There will always be edge cases where people drop things in places where they shouldn't.
The current approach doesn't handle anything that actually spawns the entity itself and doesn't use the helper.
At this point we just have to call it good enough. Throw some documentation on the spawnAdditionalDrops, or change it's name to something like onHarvested/onDropped. As vanilla has president that it's not JUST for adding more drops.